Raise an exception during save()

mayya-sharipova opened the pull request:
Raise an exception when data from Spark are saved
to a db that doesn't exist in Cloudant

Syntax formatting

BugzID: 56197

HolgerKache commented on cloudant-spark-sql/src/main/scala/com/cloudant/spark/common/JsonStoreDataAccess.scala:
```diff
@@ -44,176 +37,186 @@ import scala.util.Random
+        logger.info(s"Save total ${end-start}=${result.length} successful=$isSuccessful")
+        if (!isSuccessful)
+           throw new RuntimeException("Database " + config.getDbname() +
+            " does't exist or errors with saving data!")
```
" does not exist or is not accessible. Failed to save data!"

HolgerKache commented:
+1 after that small change. Tests look good:

Use dbName=airportcodemapping_df, indexName=null, jsonstore.rdd.partitions=5, jsonstore.rdd.maxInPartition=-1, jsonstore.rdd.minInPartition=10, jsonstore.rdd.requestTimeout=100000,jsonstore.rdd.concurrentSave=-1,jsonstore.rdd.bulkSize=1
Exception in thread "main" java.lang.RuntimeException: Database airportcodemapping_df does't exist or errors with saving data!
    at com.cloudant.spark.common.JsonStoreDataAccess$$anonfun$saveAll$1.apply$mcVI$sp(JsonStoreDataAccess.scala:204)

Commit:
Raise an exception during save() errors

Raise an exception when data from Spark are saved
to a db that doesn't exist in Cloudant

Syntax formatting

BugzID: 56197
